tests/RemoteDriverTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace RonasIT\Support\Tests;
+
+use RonasIT\Support\AutoDoc\Drivers\RemoteDriver;
+use RonasIT\Support\Tests\Support\Traits\MockTrait;
+use Illuminate\Contracts\Filesystem\FileNotFoundException;
+
+class RemoteDriverTest extends TestCase
+{
+    use MockTrait;
+
+    protected $tmpData;
+    protected $removeDriverClass;
+    protected $tmpDocumentationFilePath;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->tmpData = $this->getJsonFixture('tmp_data');
+        $this->tmpDocumentationFilePath = __DIR__ . '/../storage/temp_documentation.json';
+
+        $this->removeDriverClass = new RemoteDriver();
+    }
+
+    public function testSaveTmpData()
+    {
+        $this->removeDriverClass->saveTmpData($this->tmpData);
+
+        $this->assertFileExists($this->tmpDocumentationFilePath);
+        $this->assertFileEquals($this->generateFixturePath('tmp_data_non_formatted.json'), $this->tmpDocumentationFilePath);
+    }
+
+    public function testGetTmpData()
+    {
+        file_put_contents($this->tmpDocumentationFilePath, json_encode($this->tmpData));
+
+        $result = $this->removeDriverClass->getTmpData();
+
+        $this->assertEquals($this->tmpData, $result);
+    }
+
+    public function testGetTmpDataNoFile()
+    {
+        $result = $this->removeDriverClass->getTmpData();
+
+        $this->assertNull($result);
+    }
+
+    public function testSaveData()
+    {
+        config(['auto-doc.drivers.remote.key' => 'mocked_key']);
+        config(['auto-doc.drivers.remote.url' => 'mocked_url']);
+
+        $mock = $this->mockCLass(RemoteDriver::class, ['makeHttpRequest']);
+
+        $mock
+            ->expects($this->once())
+            ->method('makeHttpRequest')
+            ->with('post', 'mocked_url/documentations/mocked_key', $this->tmpData, [
+                'Content-Type: application/json'
+            ])
+            ->willReturn(true);
+
+        file_put_contents($this->tmpDocumentationFilePath, json_encode($this->tmpData));
+
+        $mock->saveData();
+
+        $this->assertFileDoesNotExist($this->tmpDocumentationFilePath);
+    }
+
+    public function testSaveDataWithoutTmpFile()
+    {
+        config(['auto-doc.drivers.remote.key' => 'mocked_key']);
+        config(['auto-doc.drivers.remote.url' => 'mocked_url']);
+
+        $mock = $this->mockCLass(RemoteDriver::class, ['makeHttpRequest']);
+
+        $mock
+            ->expects($this->once())
+            ->method('makeHttpRequest')
+            ->with('post', 'mocked_url/documentations/mocked_key', null, [
+                'Content-Type: application/json'
+            ])
+            ->willReturn(true);
+
+        $mock->saveData();
```

Comment:
there is no assertion in this test

Reply:
Mock RemoteDriver class also asserting that expected method had been called expected times with expected arguments

Reply:
mocks and assertions should be separated